### ISSUE TITLE
Issue wasn't resolved despite checking the "mark as resolved" box

### ISSUE THREAD

**Issue by FArekkusu** [Codewars/codewars.com#1648]
I've opened an issue. After the author fixed it, he responded to my comment marking the issue as resolved, but the `issue` tag was still present. When I clicked `reply` to see if the "mark as resolved" checkbox was present, it was indeed there.

![image](https://user-images.githubusercontent.com/31668662/49820806-cb133c00-fd70-11e8-9390-23f09a3f9f2a.png)

[Comment link](https://www.codewars.com/kata/find-the-shifted-words/discuss#5c0fe32f8a2d877826000035).

**Comment by Voileexperiments** [comment]
I've seen this happened on very old comments in a few old katas.

**Comment by Blind4Basics** [comment]
[same here](https://www.codewars.com/kata/5376b901424ed4f8c20002b7/discuss#5c59e018d773bf436ed60133), old kata but new issue.

**Comment by Voileexperiments** [comment]
I just resolved it ;-)

**Comment by FArekkusu** [comment]
Happened again [here](https://www.codewars.com/kata/5bbb8887484fcd36fb0020ca/discuss#5c6c547c147cb0731afa710c):

![image](https://user-images.githubusercontent.com/31668662/53042598-4b4fcf00-348f-11e9-8e95-0ac35669b742.png)

Looks like the reason for this is timing out during the comment processing/publishing.

**Comment by kazk** [comment]
Thanks for the update.

> Looks like the reason for this is timing out during the comment processing/publishing.

Yeah, many of the Codewars bugs are caused by this. Something happens during a transaction and data is partially updated/created.

**Comment by error256** [comment]
https://www.codewars.com/kata/558fc85d8fd1938afb000014/discuss#5cc0d85b658d6f00268001b8
This one is unresolvable. It's been resolved more than 10 times and it's still not resolved.

**Comment by kazk** [comment]
> It's been resolved more than 10 times and it's still not resolved.

Whoa
![image](https://user-images.githubusercontent.com/639336/59151817-7b938d00-89ee-11e9-9663-78f60e7242fb.png)

This doesn't seem like the timeout issue. I'm going to take a look at the data.

**Comment by kazk** [comment]
Yeah, this one looks like a data issue. The server is saying vote is invalid.

I don't know why, but most of the Codewars' JSON APIs are set to return status code `200` even on error and response contains fields like `success: false` and `error: 422`. The worst part is that the most of the frontend code is trying to handle errors using the status code :man_shrugging:

I think the comments have "marked resolved by" because they're marked as resolved before this error (same issue as approving translation, being too optimistic).

**Comment by Chrono79** [comment]
There is a post in this discussion that I can't unmark as having spoiler content:
https://www.codewars.com/kata/52f677797c461daaf7000740/discuss#5d35df6793c4cf0022e22555 (it's the issue reported by cocoderrr)
Even when it shows the "Successfully un-flagged comment", reloading the page shows it flagged again, is it related to this problem?

**Comment by kazk** [comment]
@Chrono79 That one actually have `success: true`, but it can be related to the general issue of Codewars being too optimistic.

**Comment by Blind4Basics** [comment]
seems related to this issue, so posting here: I just found a new version of that bug, apparently:

1. resolve the issue, request seems to timeout during the process
2. suggestion is neither resolved (somewhat the usual bug) nor updated with my answer (that's the new part) while the dashboard is showing my message/resolution (see below)
3. posting again, both my messages are appearing.

cheers
![image](https://user-images.githubusercontent.com/32236948/61949305-b35c9100-afaa-11e9-9214-def6b2432631.png)

**Comment by FArekkusu** [comment]
The `comment not being visible in the comment thread` and `2 comments appearing at once` are not new. And IIRC I've already encountered the `but comment is visible on the dashboard` long time ago.

**Comment by javatlacati** [comment]
That hapened to me just today

![image](https://user-images.githubusercontent.com/1063901/62386398-5702ef80-b51d-11e9-9cc0-7bc2a3e14bbd.png)

**Comment by Steffan153** [comment]
Another unresolvable one :confused: I resolved it 10 times and it wouldn't resolve: https://www.codewars.com/kata/52f677797c461daaf7000740/discuss#5cf87ce1aa824b0014c18368

**Comment by Steffan153** [comment]
I tried to resolve two comments [here](https://www.codewars.com/kata/sum-strings-as-numbers/discuss/cpp) and both were unresolvable.
An issue with that kata?

**Comment by FArekkusu** [comment]
I've just tried to do resolve the same suggestions as @Steffan153 did, and I received the following response: `{"success":false,"message":"Comment Owners is invalid","errors":[{"reason":"Comment Owners is invalid","fields":["comment_owners"]}],"error":422}`.

**Comment by error256** [comment]
I can't post a comment at this translation: https://www.codewars.com/kumite/588b21f60119130c1100008b?sel=588b4ef2ca66a34c1c000014. The response is the same "Comment Owners is invalid", it looks like it's not only about resolving issues and suggestions.

**Comment by kazk** [comment]
I've fixed the data for the mentioned comments and resolved them. The root cause is unknown, but the data stored was "invalid" and that was causing to fail to update.

If you run into this again, please post here with the URL of the comment like https://www.codewars.com/kata/52f677797c461daaf7000740/discuss#5cf87ce1aa824b0014c18368

I haven't figured out @error256's case.

**Comment by ejini6969** [comment]
It happened again today at this kata https://www.codewars.com/kata/5805ed25c2799821cb000005/discuss#61a78af81b3bc90034815217 (might be a cache issue)

**Comment by kazk** [comment]
Done. The comment's data was invalid again. Still don't know how it happens.
